src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -728,6 +728,7 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
             if (hasNaturalType)
             {
                 tupleTypeOpt = TupleTypeSymbol.Create(node.Location, elements, locations, elementNamesArray, this.Compilation, node, diagnostics);
+                Binder.ReportDiagnosticsIfObsoleteInternal(diagnostics, tupleTypeOpt, node, tupleTypeOpt.ContainingType, BinderFlags.None);
```

Comment:
Thanks. There was indeed a bug with nested case.
Although it does create duplicate warnings, putting the code in `TupleTypeSymbol.Create` did work out better. I don't think I want to pass a new flag through to avoid such duplication, given how edge case this is.
I pushed a commit with this change.